Sources/RealDeviceMap/Webhook/WebHookRequestHandler.swift

```diff
@@ -83,12 +97,16 @@ class WebHookRequestHandler {
 
         let json: [String: Any]
         do {
-            guard var jsonOpt = try (request.postBodyString ?? "").jsonDecode() as? [String: Any] else {
```

Comment:
Add 

isMadData = request.header(.origin) != nil

here instead of as global

Reply:
isMadData needs to be initialized outside of the do loop as it's referenced again on line 217 by skoodats contribution to allow for parsing of MAD encounter responses.
`} else if method == 102 && trainerLevel >= 30 || method == 102 && isMadData == true {`